pkg/sessionctx/stmtctx/stmtctx.go

```diff
@@ -443,17 +443,17 @@ func NewStmtCtxWithTimeZone(tz *time.Location) *StatementContext {
 		ctxID: stmtCtxIDGenerator.Add(1),
 	}
 	sc.typeCtx = types.NewContext(types.DefaultStmtFlags, tz, sc)
-	sc.initErrCtx()
+	sc.errCtx = newErrCtx(sc.typeCtx, errctx.LevelMap{}, sc)
 	return sc
 }
 
 // Reset resets a statement context
 func (sc *StatementContext) Reset() {
 	*sc = StatementContext{
-		ctxID:   stmtCtxIDGenerator.Add(1),
-		typeCtx: types.NewContext(types.DefaultStmtFlags, time.UTC, sc),
+		ctxID: stmtCtxIDGenerator.Add(1),
 	}
-	sc.initErrCtx()
+	sc.typeCtx = types.NewContext(types.DefaultStmtFlags, time.UTC, sc)
+	sc.errCtx = newErrCtx(sc.typeCtx, errctx.LevelMap{}, sc)
```

Comment:
Thanks for this modification. It shows the dependency of `errCtx` and it's much clearer.